src/mode.c

```diff
@@ -20,10 +21,23 @@ void init_mode(void)
   _armed_state = DISARMED;
 }
 
-void arm(void)
+bool arm(void)
 {
-  _armed_state = ARMED;
-  LED1_ON;
+  static bool started_gyro_calibration = false;
+  if (!started_gyro_calibration && _armed_state == DISARMED)
+  {
+    start_gyro_calibration();
```

Comment:
Cool, sounds good. Yeah we can clean this up when do a better state machine. I guess we already have an issue for improving the state machine (#28) so this should be captured by that, right?